test/platform_test/platform_test.go

```diff
@@ -20,363 +20,283 @@ import (
 	"net/http"
 	"testing"
 
+	. "github.com/onsi/gomega"
+
+	"github.com/Peripli/service-manager/test"
+
 	"github.com/Peripli/service-manager/test/common"
 	. "github.com/onsi/ginkgo"
 )
 
-type object = common.Object
-type array = common.Array
-
 // TestPlatforms tests for platform API
 func TestPlatforms(t *testing.T) {
+	RegisterFailHandler(Fail)
 	RunSpecs(t, "Platform API Tests Suite")
 }
 
-var _ = Describe("Service Manager Platform API", func() {
-	var ctx *common.TestContext
-
-	BeforeSuite(func() {
-		ctx = common.NewTestContext(nil)
-	})
-
-	AfterSuite(func() {
-		ctx.Cleanup()
-	})
-
-	BeforeEach(func() {
-		common.RemoveAllPlatforms(ctx.SMWithOAuth)
-	})
-
-	Describe("GET", func() {
-		Context("Missing platform", func() {
-			It("returns 404", func() {
-				ctx.SMWithOAuth.GET("/v1/platforms/999").
-					Expect().
-					Status(http.StatusNotFound).
-					JSON().Object().Keys().Contains("error", "description")
+var _ = test.DescribeTestsFor(test.TestCase{
+	API:            "/v1/platforms",
+	SupportsLabels: false,
+	SupportedOps: []test.Op{
+		test.Get, test.List, test.Delete, test.DeleteList,
+	},
+	RndResourceBlueprint:                      blueprint(true),
+	RndResourceWithoutNullableFieldsBlueprint: blueprint(false),
+	AdditionalTests: func(ctx *common.TestContext) {
+		Context("non-generic tests", func() {
+			BeforeEach(func() {
+				common.RemoveAllPlatforms(ctx.SMWithOAuth)
 			})
-		})
-
-		Context("Existing platform", func() {
-			It("returns the platform with given id", func() {
-				platform := common.MakePlatform("platform1", "cf-10", "cf", "descr")
-				reply := ctx.SMWithOAuth.POST("/v1/platforms").WithJSON(platform).
-					Expect().Status(http.StatusCreated).JSON().Object()
-				id := reply.Value("id").String().Raw()
 
-				reply = ctx.SMWithOAuth.GET("/v1/platforms/" + id).
-					Expect().
-					Status(http.StatusOK).
-					JSON().Object()
-
-				platform["id"] = id
-				common.MapContains(reply.Raw(), platform)
-			})
-		})
-	})
-
-	Describe("List", func() {
-		Context("With no platforms", func() {
-			It("returns empty array", func() {
-				ctx.SMWithOAuth.GET("/v1/platforms").
-					Expect().
-					Status(http.StatusOK).
-					JSON().Object().Value("platforms").Array().Empty()
+			Describe("POST", func() {
+				Context("With invalid content type", func() {
```

Comment:
this could be generic test also

Reply:
Post and patch i didnt have time for so we could do them later